rfcs/proposed/dynamic_selection_customization/README.md

```diff
@@ -0,0 +1,436 @@
+a manditory template argument and this can be used to set the resource type in the backend.
+
+Is it therefore becomes unnecessary to explicitly provide the resource type in the default backend.
+For example, in the code below a `round_robin_policy` is constructed by passing a vector of
+pointers to `tbb::task_group`.  The type of the resource `tbb::task_group *` can be deduce and
+passed as a template argument to the default backend. 
+
+```cpp
+    tbb::task_group t1, t2;
+    ex::round_robin_policy p{ { &t1, &t2 } };
+```
+
+If the policy is default constructed or constructed for deferred initialization, the
+resource type must be given explicitly.
+
+```cpp
+    ex::round_robin_policy<tbb::task_group*> p1{ };
+    ex::round_robin_policy<tbb::task_group*> p2{ deferred_initialization_t };
+```
+#### Default `wait_t<T>`
+
+The `wait_t<T>` can only be deduced at the time a user function is passed to submit since
+it is the type returned by the user's function. There is no way to generally determine a
+`wait_t<T>` from an arbitrary `resource_t<T>` and therefore the default backend cannot
+provide a meaningful `wait_t<T>`.
+
+#### Default `b.submit(s, f, args…)`
+
+The main role that `submit` plays in a backend is to add instrumentation around the call to `f`.
+Experience has shown us that most backends perform four basic steps in their implementation
+of the `submit` function:
+
+1. Do any setup needed for implementing reporting before calling `f`.
+2. The function `f` is called and it return value captured.
+3. Do any setup needed for implementing reporting after calling `f`, perhaps using what was returned by `f`.
+4. The *Submission* object, which typically wraps what is returned by `f`, is constructed and returned.
+
+It is not possible for a default backend to properly instrument execution for an unkown resource type to provide
+reporting of `task_time`, `task_submission` and `task_completion`. And so a default implementation
+cannot provide useful implementations for step 1 and 3. However, it can provide more finer-grained hooks
+that can be overridden to provide steps 1 and 3, without requiring a custom backend to reimplemnted the entire
+four step pattern.
+
+We propose that the default backend provide a `submit` function that implements the four step pattern
+but also calls `instrument_before_impl` and `instrument_after_impl` functions that can be overridden
+to add instrumentation by backends that need it.
+
+```cpp
+    template <typename SelectionHandle, typename Function, typename... Args>
+    auto submit(SelectionHandle s, Function&& f, Args&&... args) {
+        instrument_before_impl(s); // do insrumentation before calling `f`
+
+        return instrument_after_impl( // do insrumentation before calling `f`, and create Submission type
+            s,
+            std::forward<Function>(f)(unwrap(s), std::forward<Args>(args)...) // call f
```

Comment:
nitpick:
for explanation, rearrange code to allow function call to appear above `instrument_after_impl`